src/SourceBrowser/src/SourceIndexServer/Helpers.cs

```diff
@@ -3,93 +3,27 @@
-                {
-                    switch (key.ToLower())
-                    {
-                        // We shouldn't copy any of these request headers
-                        case "host":
-                        case "authorization":
-                        case "cookie":
-                        case "content-length":
-                        case "content-type":
-                            continue;
-                        default:
-                            req.Headers.TryAddWithoutValidation(key, values.ToArray());
-                            break;
-                    }
-                }
-
-                configureRequest?.Invoke(req);
-
-                HttpResponseMessage res = await client.SendAsync(req, HttpCompletionOption.ResponseHeadersRead).ConfigureAwait(false);
-                context.Response.RegisterForDispose(res);
-
-                foreach (var (key, values) in res.Headers)
-                {
-                    switch (key.ToLower())
-                    {
-                        // Remove headers that the response doesn't need
-                        case "set-cookie":
-                        case "x-powered-by":
-                        case "x-aspnet-version":
-                        case "server":
-                        case "transfer-encoding":
-                        case "access-control-expose-headers":
-                        case "access-control-allow-origin":
-                            continue;
-                        default:
-                            if (!context.Response.Headers.ContainsKey(key))
-                            {
-                                context.Response.Headers[key] = values.ToArray();
-                            }
-
-                            break;
-                    }
-                }
-
-                context.Response.StatusCode = (int)res.StatusCode;
-                if (res.Content != null)
-                {
-                    foreach (var (key, values) in res.Content.Headers)
-                    {
-                        if (!context.Response.Headers.ContainsKey(key))
-                        {
-                            context.Response.Headers[key] = values.ToArray();
-                        }
-                    }
-
-                    using (var data = await res.Content.ReadAsStreamAsync().ConfigureAwait(false))
-                    {
-                        await data.CopyToAsync(context.Response.Body).ConfigureAwait(false);
-                    }
-                }
+                await data.CopyToAsync(context.Response.Body).ConfigureAwait(false);
             }
         }
 
-        private static readonly HttpClient s_client = new HttpClient(new HttpClientHandler() { CheckCertificateRevocationList = true });
-
-        private static async Task<bool> UrlExistsAsync(string proxyRequestUrl)
+        private static async Task<bool> UrlExistsAsync(string proxyRequestPath)

```

Comment:
NIT: Does this need the async modifier? I don't see any awaits in the body.

Reply:
@directhex already mentioned he's fixing that: https://github.com/dotnet/source-indexer/pull/154#issue-2381291935

Reply:
Oops missed that, sorry